tests/app/views/test_services.py

```diff
@@ -490,6 +493,197 @@ def test_should_400_if_mismatched_service_id(self):
+
+
+class TestShouldCallSearchApiOnPost(BaseApplicationTest):
+    def setup(self):
+        super(TestShouldCallSearchApiOnPost, self).setup()
+        now = datetime.now()
+        payload = self.load_example_listing("G6-IaaS")
+        with self.app.app_context():
+            db.session.add(
+                Framework(id=1, expired=False, name="G-Cloud 6")
+            )
+            db.session.add(
+                Supplier(supplier_id=1, name=u"Supplier 1")
+            )
+            db.session.add(Service(service_id="1234567890123456",
+                                   supplier_id=1,
+                                   updated_at=now,
+                                   status='published',
+                                   created_at=now,
+                                   updated_by="tests",
+                                   framework_id=1,
+                                   updated_reason="test data",
+                                   data=payload))
+            db.session.commit()
+
+    def test_should_index_on_service_post(self):
+        with self.app.app_context():
+            search_api_client.index = Mock(return_value=True)
+
+            payload = self.load_example_listing("G6-IaaS")
+            payload['id'] = "1234567890123456"
+            self.client.post(
+                '/services/1234567890123456',
+                data=json.dumps(
+                    {
+                        'update_details': {
+                            'updated_by': 'joeblogs',
+                            'update_reason': 'whateves'},
+                        'services': payload}
+                ),
+                content_type='application/json')
+
+            service = Service.query.filter(Service.service_id ==
+                                           "1234567890123456").first()
+            search_api_client.index.assert_called_with(
+                "1234567890123456",
+                service.data,
+                "Supplier 1"
+            )
+
+    def test_should_not_index_on_service_post_if_db_exception(self):
+        with self.app.app_context():
+            search_api_client.index = Mock(return_value=True)
+            c = db.session.commit
+            db.session.commit = Mock(
+                side_effect=IntegrityError(
+                    'message', 'statement', 'params', 'orig'))
+
+            payload = self.load_example_listing("G6-IaaS")
+            payload['id'] = "1234567890123456"
+            try:
+                self.client.post(
+                    '/services/1234567890123456',
+                    data=json.dumps(
+                        {
+                            'update_details': {
+                                'updated_by': 'joeblogs',
+                                'update_reason': 'whateves'},
+                            'services': payload}
+                    ),
+                    content_type='application/json')
+            except IntegrityError:
```

Comment:
It's testing a DB commit failure does not subsequently lead to an indexing of that change in ES - so no calls to the API. As the POST method does not handle integrity exceptions - it's a 500 - I need to catch the exception to make the assertion on my mock.

Talked with Rob, and I'm going to add integrity checking to the POST method, as it does on the PUT method.